core/capabilities/ccip/configs/solana/chain_writer.go

```diff
@@ -150,49 +157,61 @@ func getExecuteMethodConfig(fromAddress string, routerProgramAddress string, com
 						InternalField: chainwriter.InternalField{
 							TypeName: "TokenAdminRegistry",
 							Location: "LookupTable",
+							// TokenAdminRegistry is in the router program so need to provide the router's IDL
+							IDL: ccipRouterIDL,
 						},
 					},
 				},
-			},
-			StaticLookupTables: []solana.PublicKey{
-				commonAddressesLookupTable,
+				getCommonAddressLookupTableConfig(offrampProgramAddress),
 			},
 		},
 		Accounts: []chainwriter.Lookup{
-			getRouterAccountConfig(routerProgramAddress),
+			getOfframpAccountConfig(offrampProgramAddress),
+			getReferenceAddressesConfig(offrampProgramAddress),
 			chainwriter.PDALookups{
-				Name: "SourceChainState",
-				PublicKey: chainwriter.AccountConstant{
-					Address: routerProgramAddress,
-				},
+				Name:      "SourceChainState",
+				PublicKey: offrampAddressConstant(offrampProgramAddress),
 				Seeds: []chainwriter.Seed{
 					{Static: []byte("source_chain_state")},
-					{Dynamic: chainwriter.AccountLookup{Location: destChainSelectorPath}},
+					{Dynamic: chainwriter.AccountLookup{Location: sourceChainSelectorPath}},
 				},
 				IsSigner:   false,
 				IsWritable: false,
 			},
 			chainwriter.PDALookups{
-				Name: "CommitReport",
-				PublicKey: chainwriter.AccountConstant{
-					Address: routerProgramAddress,
-				},
+				Name:      "CommitReport",
+				PublicKey: offrampAddressConstant(offrampProgramAddress),
 				Seeds: []chainwriter.Seed{
-					{Static: []byte("external_execution_config")},
-					{Dynamic: chainwriter.AccountLookup{Location: destChainSelectorPath}},
+					{Static: []byte("commit_report")},
+					{Dynamic: chainwriter.AccountLookup{Location: sourceChainSelectorPath}},
 					{Dynamic: chainwriter.AccountLookup{
 						// The seed is the merkle root of the report, as passed into the input params.
-						Location: "Info.MerkleRoots.MerkleRoot",
+						Location: merkleRoot,
 					}},
 				},
 				IsSigner:   false,
 				IsWritable: true,
 			},
+			chainwriter.AccountConstant{
+				Name:       "Offramp",
+				Address:    offrampProgramAddress,
+				IsSigner:   false,
+				IsWritable: false,
+			},
 			chainwriter.PDALookups{
-				Name: "ExternalExecutionConfig",
-				PublicKey: chainwriter.AccountConstant{
-					Address: routerProgramAddress,
+				Name:      "AllowedOfframp",
+				PublicKey: getRouterConfig(offrampProgramAddress),
+				Seeds: []chainwriter.Seed{
+					{Static: []byte("allowed_offramp")},
+					{Dynamic: chainwriter.AccountLookup{Location: sourceChainSelectorPath}},
+					{Dynamic: offrampAddressConstant(offrampProgramAddress)},
 				},
```

Comment:
I see in the `context.rs` the corresponding seed for this is `offramp.key().as_ref()`. Just want to confirm that that is indeed the program address... don't have much experience working with anchor contracts so just wanted to be sure

Reply:
I think that's the case since other parts of the code use `.key()` to refer to the Pubkey